### PR TITLE
Add :var header argument

### PR DESCRIPTION
**Pull request by yunmikun2**
It supports lists strings and integer data types.

### CODE REVIEW

File: ob-elixir.el [hunk 1]
```diff
@@ -46,11 +47,38 @@
...
     (with-temp-file tmp (insert body))
+    (ob-elixir-eval session (string-join var-lines "\n"))
     (ob-elixir-eval session (format "import_file(\"%s\")" tmp))))
 
+(defun ob-elixir--variable-assignments (params)
+  (let ((vars (org-babel--get-vars params)))
+    (mapcar #'ob-elixir--var-to-assignment vars)))
+
+(defun ob-elixir--var-to-assignment (var)
+  (format "%s = %s"
```
Comment (sirikid) [inline ob-elixir.el]: There should be newline

File: ob-elixir.el [hunk 2]
```diff
@@ -46,11 +47,38 @@
...
+
+    ((numberp data)
+     (format "%d" data))
+
+    ((stringp data)
+     (let* ((with-escaped-backslashes
+                (replace-regexp-in-string "\\\\" "\\\\" data))
+            (with-escaped-quotes
+                (replace-regexp-in-string "\"" "\\\\\"" with-escaped-backslashes)))
+       (format "\"%s\"" with-escaped-quotes)))))
```
Comment (sirikid) [inline ob-elixir.el]: Perfect case for `thread-last` from `subr-x`

File: ob-elixir.el [hunk 3]
```diff
@@ -46,11 +47,38 @@
 
 (defun org-babel-execute:elixir (body params)
   (let ((session (cdr (assoc :session params)))
-        (tmp (org-babel-temp-file "elixir-")))
+        (tmp (org-babel-temp-file "elixir-"))
+        (var-lines (ob-elixir--variable-assignments params)))
```
Comment (sirikid) [inline ob-elixir.el]: Just `vars` would be fine

File: ob-elixir.el (hunk 2 above)
Comment (sirikid) [inline ob-elixir.el]: Cases order: numbers, strings, lists

Comment (yunmikun2) [reply ob-elixir.el]: Why does it matter?

File: ob-elixir.el (hunk 1 above)
Comment (yunmikun2) [reply ob-elixir.el]: But I do it in a different place -- in `org-babel-execute:elixir`.

File: ob-elixir.el (hunk 3 above)
Comment (yunmikun2) [reply ob-elixir.el]: I was just looking at emacs-lisp's execute function and vars were called like that there. I'll change the name.

File: ob-elixir.el (hunk 2 above)
Comment (yunmikun2) [reply ob-elixir.el]: Huh, I was dreaming about this function in lisp for so long. Thanks for pointing this out for me.

Comment (sirikid) [reply ob-elixir.el]: Good manners: short branches before long, non-recursive before recursive

**Pull request merged**